docs/guide/vite.md

```diff
@@ -0,0 +1,383 @@
+# Vite
+
+[Vite](https://vite.dev/) is a modern build tool that significantly improves the development experience. Unlike traditional bundlers like webpack, Vite leverages native ES modules in the browser to enable lightning-fast hot module replacement (HMR) and on-demand compilation. During development, this means your changes appear instantly in the browser without rebuilding the entire bundle. When building for production, Vite uses Rollup to create highly optimized assets.
```

Comment:
The instant HMR described here happens only when the Framework supports it. A vanilla JS doesn't do that, it does a browser page reload (A manual "hot" is possible, described in the vite docs). I think we need a clarfiication here - when your UI framework by choice supports HMR (all major frameworks does that).

Reply:
Changes made in the first three paragraphs. Hopefully it is clearer and more accurate.

Reply:
Changes made in the first three paragraphs. Hopefully it is clearer and more accurate.